tests/integration/test_integration.py

```diff
@@ -29,209 +31,351 @@
-            f"{SELF_SIGNED_CERTIFICATES_APPLICATION_NAME}/0"
-        ]
-        assert isinstance(self_signed_certificates_unit, Unit)
-        action = await self_signed_certificates_unit.run_action(
-            action_name="get-ca-certificate",
+        unit = ops_test.model.units[f"{APPLICATION_NAME}/1"]
+        assert isinstance(unit, Unit)
+        k8s_namespace = ops_test.model.name
+        crash_pod(name=f"{APPLICATION_NAME}-1", namespace=k8s_namespace)
+        await ops_test.model.wait_for_idle(
+            apps=[APPLICATION_NAME],
+            status="active",
+            timeout=1000,
+            wait_for_exact_units=NUM_VAULT_UNITS,
         )
-        return await ops_test.model.get_action_output(action_uuid=action.entity_id, wait=timeout)
 
     @pytest.mark.abort_on_fail
-    @pytest.fixture(scope="module")
-    async def deploy_prometheus(self, ops_test: OpsTest) -> None:
-        """Deploy Prometheus.
-
-        Args:
-            ops_test: Ops test Framework.
-        """
+    async def test_given_application_is_deployed_when_scale_up_then_status_is_active(
+        self, ops_test: OpsTest, deploy_phase_1
+    ):
         assert ops_test.model
-        await ops_test.model.deploy(
-            "prometheus-k8s",
-            application_name=PROMETHEUS_APPLICATION_NAME,
-            trust=True,
+        num_units = 5
+        app: Application = ops_test.model.applications[APPLICATION_NAME]
+        await app.scale(num_units)
+
+        await ops_test.model.wait_for_idle(
+            apps=[APPLICATION_NAME],
+            status="active",
+            timeout=1000,
+            wait_for_exact_units=num_units,
         )
 
     @pytest.mark.abort_on_fail
-    @pytest.fixture(scope="module")
-    async def deploy_loki(self, ops_test: OpsTest) -> None:
-        """Deploys Loki.
+    async def test_given_application_is_deployed_when_scale_down_then_status_is_active(
+        self, ops_test: OpsTest, deploy_phase_1
+    ):
+        assert ops_test.model
+        app: Application = ops_test.model.applications[APPLICATION_NAME]
+        await app.scale(NUM_VAULT_UNITS)
 
-        Args:
-            ops_test: Ops test Framework.
-        """
+        await ops_test.model.wait_for_idle(
+            apps=[APPLICATION_NAME],
+            status="active",
+            timeout=1000,
+            wait_for_exact_units=NUM_VAULT_UNITS,
+        )
+
+    @pytest.mark.abort_on_fail
+    async def test_given_vault_kv_requirer_deployed_when_vault_kv_relation_created_then_status_is_active(
+        self, ops_test: OpsTest, deploy_phase_2
+    ):
         assert ops_test.model
-        await ops_test.model.deploy(
-            "loki-k8s",
-            application_name=LOKI_APPLICATION_NAME,
-            trust=True,
+        await ops_test.model.integrate(
+            relation1=f"{APPLICATION_NAME}:vault-kv",
+            relation2=f"{VAULT_KV_REQUIRER_APPLICATION_NAME}:vault-kv",
+        )
+        await ops_test.model.wait_for_idle(
```

Comment:
Do we depend on the status of `VAULT_KV_REQUIRER_APPLICATION_NAME` being active? If not, I'd not check whether it becomes active, we don't have much control over that status and we are not testing it.
I know this might have been already in the tests before this PR, but I think it's worth thinking about.

Reply:
I know we are using a test charm for it, so if we depend on the status it's fine :)